### PR TITLE
[G7T5-80, G7T5-85] Add create job page for HR

### PR DESCRIPTION
**Pull request by jonathantan1425**
# Description

Fixes [Jira ticket #G7T5-80](https://is212g7t5.atlassian.net/browse/G7T5-80)

image
image
image

## Type of change

Adds create new job page for HR under `/create-job`. Only users logged in as HR can access the page, while the rest will encounter an error message
image

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [x] I have added the corresponding testcases into the [testcase document](https://docs.google.com/spreadsheets/d/1QOyUN_kFN0fddLkjAQz2DK3jou3cWdN9CJsNbl3v0Kg/edit#gid=0)

Please indicate the testcase ID you have added or are using

- [x] G7T5-1-1
- [x] G7T5-1-2
- [x] G7T5-1-3
- [x] G7T5-1-4
- [x] G7T5-1-5
- [x] G7T5-1-6
- [x] G7T5-1-7
- [x] G7T5-1-8
- [x] G7T5-1-9
- [x] G7T5-1-10
- [x] G7T5-1-11
- [x] G7T5-1-19
- [x] G7T5-1-20
- [x] G7T5-1-21
- [x] G7T5-1-22

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/components/job/hr/CreateJob.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    } else {
+      setErrors([]);
+      setDisplayPopup(true);
+    }
+  }
+
+  const handleJobNameChange = (e) => {
+    if (e.target.value.length <= 50) {
+      setJobName(e.target.value);
+      e.preventDefault();
```
Comment (quinncheong) [inline app/src/components/job/hr/CreateJob.jsx:36]: Should be outside the for loop?

File: app/src/api/jobs.js [hunk 2]
```diff
@@ -28,6 +28,21 @@ export const getAllJobsAndSkills = async () => {
   }
 };
 
+export const createJob = async (jobName, jobDesc) => {
```
Comment (quinncheong) [inline app/src/api/jobs.js:31]: Maybe cna rename to createJobWithJobNameAndDesc

**Review by quinncheong — APPROVED** [review]
Merge if you think its ok

File: app/src/api/jobs.js (hunk 2 above)
Comment (jonathantan1425) [reply app/src/api/jobs.js:31]: I think don't need to be so verbose for this? cos the only way to create a job is with name and desc

**Pull request merged**